service/src/main/kotlin/fi/espoo/evaka/daycare/service/AbsenceQueries.kt

```diff
@@ -167,6 +167,28 @@ fun Database.Transaction.deleteChildAbsences(childId: ChildId, date: LocalDate):
         .executeAndReturnGeneratedKeys()
         .toList<AbsenceId>()
 
+fun Database.Transaction.deleteNonSystemGeneratedAbsencesByCategoryInRange(
+    childId: ChildId,
+    range: DateRange,
+    category: AbsenceCategory
```

Comment:
This could take a list of absence categories: `categories: List<AbsenceCategory>`.

The list can be used in the query like this: `... AND category = ANY (:categories::absence_category[]) ...`